packages/test/test-end-to-end-tests/src/test/gc/gcTrailingOps.spec.ts

```diff
@@ -0,0 +1,177 @@
+            assert(dataStoreTimestamp1 === undefined, `Should have referenced datastore`);
+            assert(blobTimestamp1 === undefined, `Should have referenced blob`);
+
+            // Create trailing ops where the datastore and blob are unreferenced
+            mainDefaultDataStore._root.delete("datastore");
+            mainDefaultDataStore._root.delete("blob");
+            await provider.ensureSynchronized();
+
+            mainContainer.close();
+            mainSummarizer.close();
+
+            // Load a new container/summarizer from the summary and trailing ops
+            const summarizer = await createSummarizer(provider, mainContainer, summary1.summaryVersion, gcOptions, configProvider);
+
+            // Ensure trailing ops are processed, summarize, and verify that the datastore and blob are unreferenced
+            await provider.ensureSynchronized();
+            const summary2 = await summarizeNow(summarizer);
+            const unreferencedTimestamps2 = await getUnreferencedTimestamps(summary2.summaryTree);
+            const dataStoreTimestamp2 = unreferencedTimestamps2.get(newDataStore.entryPoint.absolutePath.slice(1));
+            const blobTimestamp2 = unreferencedTimestamps2.get(blobHandle.absolutePath.slice(1));
+            assert(dataStoreTimestamp2 !== undefined, `Should have unreferenced datastore`);
+            assert(blobTimestamp2 !== undefined, `Should have unreferenced blob`);
+        });
+
+        // Note: we should not need an itExpects here as this error should not have fired.
+        itExpects(`A summary has a datastore and blob unreferenced, but trailing ops referenced them ${tombstoneEnabled ? "after sweep timeout" : "before sweep timeout"}`,
+        tombstoneEnabled ? [
+            { eventName: "fluid:telemetry:Summarizer:Running:SweepReadyObject_Revived" },
+            { eventName: "fluid:telemetry:Summarizer:Running:SweepReadyObject_Revived" },
+        ] : [],
+        async () => {
+            const mainContainer = await provider.makeTestContainer(testContainerConfig);
+            const mainDefaultDataStore = await requestFluidObject<ITestDataObject>(mainContainer, "default");
+            await waitForContainerConnection(mainContainer);
+
+            // Create a data store and blob.
+            const newDataStore = await mainDefaultDataStore._context.containerRuntime.createDataStore(TestDataObjectType);
+            assert(newDataStore.entryPoint !== undefined, `Should have a handle`);
+            const blobContents = "Blob contents";
+            const blobHandle = await mainDefaultDataStore._runtime.uploadBlob(stringToBuffer(blobContents, "utf-8"));
+
+            // Create a summarizer
+            const mainSummarizer = await createSummarizer(provider, mainContainer, undefined, gcOptions, configProvider);
+
+            // Create trailing ops where the datastore and blob are unreferenced
```

Comment:
These are not the trailing ops. They are after summary1 happens right.

Reply:
Woops, comment is actually in the wrong part